src/dict.c

```diff
@@ -778,6 +790,11 @@ dictEntry *dictFindByHash(dict *d, const void *key, const uint64_t hash) {
     /* Rehash the hash table if needed */
     _dictRehashStepIfNeeded(d,idx);
 
+    /* Check if we can use the compare function with length to avoid recomputing length of key always */
+    keyCmpFuncWithLen cmpFuncWithLen = dictGetKeyCmpFuncWithLen(d);
+    keyLenFunc keyLenFunc = dictGetKeyLenFunc(d);
+    const int has_len_fn = (keyLenFunc!=NULL && cmpFuncWithLen != NULL);
+    const size_t keyLen = has_len_fn ? keyLenFunc(d,key) : 0;
```

Comment:
```suggestion
    const size_t key_len = has_len_fn ? keyLenFunc(d,key) : 0;
```
he_key is snake, it feels better to unify it.